spec/features/work_packages/share_spec.rb

```diff
@@ -282,4 +282,46 @@ def inherited_member_roles(group:)
       share_modal.expect_no_invite_option
     end
   end
+
+  context 'when having global invite permission' do
+    shared_let(:global_manage_user) { create(:user, global_permissions: %i[manage_user create_user]) }
+    let(:current_user) { global_manage_user }
+
+    it 'allows inviting and directly sharing with a user who is not part of the instance yet' do
+      work_package_page.visit!
+      click_button 'Share'
+
+      share_modal.expect_open
+      share_modal.expect_shared_count_of(6)
+
+      share_modal.create_and_invite_user('hello@world.de', 'View')
+
+      share_modal.expect_shared_count_of(7)
+
+      new_user = User.last
+      share_modal.expect_shared_with(new_user, 'View', position: 1)
+
+      perform_enqueued_jobs
+      # Only one combined email for create and share should be send out
+      expect(ActionMailer::Base.deliveries.size).to eq(1)
```

Comment:
How about adding two more expectations that after being invited, the new user can be interacted with in the modal just as any other by:

1. Updating its Role inline
2. Updating its Role via the Form (and no longer showing up as "needs to be invited")
3. Removing its share

```ruby
      share_modal.change_role(new_user, 'Comment')
      share_modal.expect_shared_with(new_user, 'Comment', position: 1)
      share_modal.expect_shared_count_of(7)

      share_modal.invite_user(new_user, 'Edit')
      share_modal.expect_shared_with(new_user, 'Edit', position: 1)
      share_modal.expect_shared_count_of(7)
      
      share_modal.remove_user(new_user)
      share_modal.expect_not_shared_with(new_user)
      share_modal.expect_shared_count_of(6)
```